src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -1471,6 +1495,27 @@ theorem mul_twoPow_eq_shiftLeft (x : BitVec w) (i : Nat) :
       apply Nat.pow_dvd_pow 2 (by omega)
     simp [Nat.mul_mod, hpow]
 
+@[simp]
+theorem and_twoPow_eq_getLsb {x : BitVec w} {i : Nat} :
+    x &&& (twoPow w i) = if x.getLsb i then twoPow w i else 0#w := by
+  ext j
+  simp only [getLsb_and, getLsb_twoPow]
+  by_cases hj : i = j <;> by_cases hx : x.getLsb i <;> simp_all
+
+theorem BitVec.toNat_twoPow {w : Nat} {i : Nat} : (twoPow w i).toNat = 2^i % 2^w := by
```

Comment:
This has two `BitVec`s in the name. If you remove one, you'll find this already exists.